src/commands/start-daemon.command.ts

```diff
@@ -0,0 +1,84 @@
+import { ChildProcess } from "child_process";
+import colors from "colors";
+import { existsSync } from "fs";
+import { createServer, Server, Socket } from "net";
+
+import { logsDaemonCommand } from "../daemon-command/logs.daemon-command";
+import { restartDaemonCommand } from "../daemon-command/restart.daemon-command";
+import { shutdown } from "../daemon-command/shutdown";
+import { shutdownDaemonCommand } from "../daemon-command/shutdown.daemon-command";
+import { startDaemonCommand } from "../daemon-command/start.deamon-command";
+import { statusDaemonCommand } from "../daemon-command/status.daemon-command";
+import { ScriptDefinition } from "../script-definition.type";
+
+export interface Daemon {
+    logSockets: { socket: Socket; name: string | null }[];
+    scripts: ScriptDefinition[];
+    processes: { [key: string]: ChildProcess };
+    shuttingDown: boolean;
+    server?: Server;
+}
+
+export const startDaemon = async (pmConfigFilePathOverride?: string): Promise<void> => {
+    const pmConfigFilePath = pmConfigFilePathOverride ? pmConfigFilePathOverride : "dev-pm.config.js";
+    const { scripts }: { scripts: ScriptDefinition[] } = await import(`${process.cwd()}/${pmConfigFilePath}`);
+    const processes: { [key: string]: ChildProcess } = {};
+    const logSockets: { socket: Socket; name: string | null }[] = [];
+    const daemon: Daemon = {
+        logSockets,
+        scripts,
+        processes,
+        shuttingDown: false,
+        server: undefined,
+    };
+
+    if (existsSync("./.pm.sock")) {
+        console.log(
+            "Could not start dev-pm server. A '.pm.sock' file already exists. \nThere are 2 possible reasons for this:\nA: Another dev-pm instance is already running. \nB: dev-pm crashed and left the file behind. In this case please remove the file manually.",
+        );
```

Comment:
Please use the log style used throughout the project:

```
console.log(`${colors.bgYellow.bold.black(" dev-pm ")} Could not start dev-pm server. A '.pm.sock' file already exists. \nThere are 2 possible reasons for this:\nA: Another dev-pm instance is already running. \nB: dev-pm crashed and left the file behind. In this case please remove the file manually.`);
```

Reply:
See https://github.com/vivid-planet/dev-process-manager/pull/21#discussion_r988612152 - this is a good example where it does not have any benefit